dockerfiles/base/tests/cmd_start_tests.bats

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bats
+# Copyright (c) 2012-2017 Red Hat, Inc
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#   Marian Labuda - Initial Implementation
+
+source $BATS_BASE_DIR/tests/test_base.sh
+
+# Kill running che server instance if there is any to be able to run tests
+setup() {
+  kill_running_named_container che
+  remove_named_container che
+}
+
+@test "test cli 'start' with default settings" {
+  #GIVEN
+  if [ ! port_is_free 8080 ]; then
+    [ "$status" -eq 1 ]
+    [ "$output" = "Default port 8080 for che server is used. Cannot run this test on default che server port" ]
+  fi
+  tmp_path=${TESTRUN_DIR}/start1
+
+  #WHEN
+  docker run -v $SCRIPTS_DIR:/scripts/base -v /var/run/docker.sock:/var/run/docker.sock -v $tmp_path:/data $CLI_IMAGE start
+
+  #THEN
+  [[ $(docker inspect --format="{{.State.Running}}" che) -eq "true" ]]
+  ip_address=$(docker inspect -f {{.NetworkSettings.Networks.bridge.IPAddress}} che)
+  curl -fsS http://${ip_address}:8080  > /dev/null
+}
+
+@test "test cli 'start' with custom port" {
+  #GIVEN
+  tmp_path=${TESTRUN_DIR}/start2
+  free_port=$(get_free_port)
```

Comment:
Yes, we need it. I was looking into Che and it is using default network interface docker0 (On my side 172.17.0.1). Che server runs in a container with an IP (here 172.17.0.4). Che server is accessible via port 8080. I can access to the server via 172.17.0.4:8080 but also via 172.17.0.1:8080. But the server also expose ports and forward ports to localhost. So the server is accessible also on localhost:8080. The port 8080 can be found in state LISTEN in netstat output. Same applies also to workspaces, but those take ports from 32xxx (at least now when I was running them, haven't look closer to scripts what is the algorithm to choose ports). And those are forwarded as well.

Ok now to the reason why I need free ports. First problem could be caused if there would be other docker images running and those would use port required by a che server. It would fail. Second problem is having a service running on localhost (e.g. I have running wildfly locally, which uses port 8080). And this would fail too.